src/navigation/Pagination/UsePagination.tsx

```diff
@@ -0,0 +1,51 @@
+// import * as React from "react";
+import { useMemo } from "react";
+
+export const DOTS = "...";
+
+const range = (start, end) => {
+  const length = end - start + 1;
+  return Array.from({ length }, (_, idx) => idx + start);
+};
+
+export const usePagination = ({ currentPage, maxPages }) => {
```

Comment:
**issue:** That's nice that you implemented own hook, but this hook should help us with getting `currentPageRecords`, `recordsPerPage`, and `paginatedRecords`. Your implementation should look like:

```
import { useEffect, useState } from 'react';

// Im using generics to
interface UsePaginationParams<Data> {
  records: Array<Data>;
  maxRecordsPerPage: number;
}

export const usePagination = <Data>(
  params: UsePaginationParams<Data>
) => {
  // Here you can keep paginated records
  const [paginatedRecords, setPaginatedRecords] = useState<
    Array<Data>
  >([]);

  // Here you can keep current page index
  const [currentPageIndex, setPageIndex] = useState<number>(0);

  useEffect(() => {
    // Here you can paginate your records using recordsPerPage, and set them in useState above
  }, [params.records]);

  const handleNextPage = () => {
    // Change page index to next than current
  };

  const handleSetPage = (pageIndex: number) => {
    // Change page index selected
  };

  const handlePrevPage = () => {
    // Change page index to prev from current
  };

  return {
    currentPageIndex,
    maxPages: paginatedRecords.length,
    paginatedRecords,
  };
};

```